console-framework-client/src/main/java/io/axoniq/console/framework/application/RSocketDomainEntityDataResponder.kt

```diff
@@ -0,0 +1,112 @@
+                domainEventAccessMode == DomainEventAccessMode.PREVIEW_PAYLOAD_ONLY
+
+        // Fetch domain events starting from the calculated sequence number (offset)
+        val events = domainEventStreamProvider
+                .getDomainEventStream(query.entityId, startIndex.toLong()) ?: emptyList()
+
+        // If the result is empty, return an empty result but still indicate position
+        if (events.isEmpty()) {
+            return DomainEventsResult(
+                    entityId = query.entityId,
+                    entityType = "",
+                    domainEvents = emptyList(),
+                    page = query.page,
+                    pageSize = query.pageSize,
+                    totalCount = startIndex.toLong() // Total so far is up to this offset
+            )
+        }
+
+        // Map the first 'pageSize' events into the response model
+        val pagedEvents = events
+                .asSequence()
+                .take(query.pageSize)
+                .map { event ->
+                    DomainEvent(
+                            sequenceNumber = event.sequenceNumber,
+                            timestamp = event.timestamp,
+                            payloadType = event.payloadType.toString(),
+                            payload = if (includePayload)
+                                serializer.serialize(event.payload, String::class.java).data
+                            else ""
+                    )
+                }
+                .toCollection(ArrayList())
+
+        // Total count is approximated as startIndex + number of events returned
+        val totalEvents = startIndex + events.size
+
+        return DomainEventsResult(
+                entityId = events.first().aggregateIdentifier,
+                entityType = events.first().type,
+                domainEvents = pagedEvents,
+                page = query.page,
+                pageSize = query.pageSize,
+                totalCount = totalEvents.toLong()
+        )
+    }
+
+
+    private fun handleEntityStateAtSequenceQuery(query: EntityStateAtSequenceQuery): EntityStateResult {
+        logger.debug("Handling AxonIQ Console ENTITY_STATE_AT_SEQUENCE query for request [{}]", query)
+        takeIf {
+            logger.debug("Access mode is set on: {}.", domainEventAccessMode)
+            domainEventAccessMode == DomainEventAccessMode.FULL ||
+                    domainEventAccessMode == DomainEventAccessMode.LOAD_DOMAIN_STATE_ONLY
+        }?.let {
+            domainEventStreamProvider.loadDomainStateAtSequence<String>(
+                    type = query.type,
+                    entityIdentifier = query.entityId,
+                    maxSequenceNumber = query.maxSequenceNumber)?.let {
+                return EntityStateResult(
+                        type = query.type,
+                        entityId = query.entityId,
+                        maxSequenceNumber = query.maxSequenceNumber,
+                        state = it
```

Comment:
A double `let` with overshadowing `it` generally is not the best practice. Please name the it :) 

In addition, with the extension function to truncate:

```suggestion
                        state = state.truncate(2 * 1024 * 1024) // truncate to 2MB
```